bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/DefaultClangdConfigurationVisibility.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Bachmann electronic GmbH and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
+ *******************************************************************************/
+package org.eclipse.cdt.lsp.clangd;
+
+import org.osgi.service.component.annotations.Component;
+
+@Component(property = { "service.ranking:Integer=0" })
+public class DefaultClangdConfigurationVisibility implements ClangdConfigurationVisibility {
```

Comment:
Component implementation shouldn't be a part of API, and should be final since it is not intended to be subclassed